### ISSUE TITLE
SA1127 Wrong analysis?

### ISSUE THREAD

**Issue by cryo75** [DotNetAnalyzers/StyleCopAnalyzers#2261]
I have this method which causes SA1127:

`public static IQueryable<T> Get<T>(this DbSet<T> dbSet, Expression<Func<T, bool>> predicate) where T : class`

Why is this method causing the warning? It shouldn't.

**Comment by vweijsters** [comment]
That is weird indeed. The only thing that I can think of is that you have word wrapping turned on in the editor. If that is the case, the editor will show the `where` clause to be on its own line, but the Roslyn compiler does not see / have this line break, resulting in an SA1127 diagnostic.

Can you verify that this isn't caused by word wrapping? (There is an arrow like symbol at the end of the line if word wrapping is happening)

**Comment by cryo75** [comment]
I don't have word wrapping.

**Comment by sharwell** [comment]
@cryo75 You need to move the generic type constraint for that code to not produce a warning:

```csharp
public static IQueryable<T> Get<T>(this DbSet<T> dbSet, Expression<Func<T, bool>> predicate)
    where T : class
```

SA1127 should provide a code fix that shows you the corrected code automatically. See the [SA1127 documentation](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1127.md) for more information and examples.
